dotsql_test.go

```diff
@@ -322,10 +356,136 @@ func TestQueryContext(t *testing.T) {
 	}
 }
 
+func TestQueryOptions(t *testing.T) {
+	dot := DotSql{
+		queries: map[string]*template.Template{
+			"select": createTemplate(t, "SELECT * from users WHERE name = ?{{if  .exclude_deleted}} AND deleted IS NULL{{end}}"),
+		},
+	}
+	dotExcludeDeleted := dot.WithData(map[string]any{"exclude_deleted": true})
+
+	queryerStub := func(err error) *QueryerMock {
+		return &QueryerMock{
+			QueryFunc: func(_ string, _ ...interface{}) (*sql.Rows, error) {
+				if err != nil {
+					return nil, err
+				}
+				return &sql.Rows{}, nil
+			},
+		}
+	}
+
+	arg := "test"
+
+	// query not found
+	q := queryerStub(nil)
+	rows, err := dot.Query(q, "insert", arg)
+	if rows != nil {
+		t.Error("rows expected to be nil, got non-nil")
+	}
+
+	if err == nil {
+		t.Error("err expected to be non-nil, got nil")
```

Comment:
might be pre-existing, but in these cases I think using `t.Fatal` make more sense, thoughts?